daemon/recording.c

```diff
@@ -529,13 +546,20 @@ static int rec_pcap_meta_finish_file(struct call *call) {
 	char new_metapath[prefix_len + fn_len + ext_len + 1];
 	snprintf(new_metapath, prefix_len+fn_len+1, "%s/metadata/%s", spooldir, meta_filename);
 	snprintf(new_metapath + prefix_len+fn_len, ext_len+1, ".txt");
-	return_code = return_code || rename(recording->meta_filepath, new_metapath);
-	if (return_code != 0) {
-		ilog(LOG_ERROR, "Could not move metadata file \"%s\" to \"%s/metadata/\"",
-				 recording->meta_filepath, spooldir);
+	if (!strcmp(rtpe_config.rec_method, "all")) {
+		char command[256];
+
+		sprintf(command, "cp %s %s", recording->meta_filepath, new_metapath);
+		return_code = system(command);
```

Comment:
Oof... please don't use system() to execute external shell commends. I'll have to look at this in more detail, but isn't the format of the metadata file completely different between the two methods? Are they both pointing to the same file name? If so then perhaps it would be better to make a proper distinction between the two, either in a different path, or using a different file name suffix or something... or simply forgo the `pcap` format file in favour of the `proc` format file in case of `all`.